src/javascript/crypto/e2e/openpgp/pgpmime/pgpmail.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * @license
+ * Copyright 2015 Google Inc. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview PGP/MIME message entity builder (RFC 2045, RFC 2822, RFC 3156).
+ * @author yzhu@yahoo-inc.com (Yan Zhu)
+ */
+
+goog.provide('e2e.openpgp.pgpmime.PgpMail');
+
+goog.require('e2e.openpgp.pgpmime.Constants');
+goog.require('e2e.openpgp.pgpmime.MimeNode');
+goog.require('goog.array');
+
+
+goog.scope(function() {
+var pgpmime = e2e.openpgp.pgpmime;
+var constants = pgpmime.Constants;
+
+
+
+/**
+ * Constructs an object that represents a PGP/MIME email.
+ * @param {!e2e.openpgp.pgpmime.types.ContentAndHeaders} content The encrypted
+ *     content of the email, including necessary email headers
+ * @param {string=} opt_preamble Text preceding the body of the message. This
+ *     text will be appended to the beginning of the PGP-MIME message, within
+ *     the root node
+ * @constructor
+ */
+e2e.openpgp.pgpmime.PgpMail = function(content, opt_preamble) {
+  this.content = content;
+  this.preamble = opt_preamble || '';
+};
+
+
+/**
+ * Builds a plaintext serialized MIME tree for the email.
+ * @return {string}
+ */
+e2e.openpgp.pgpmime.PgpMail.prototype.buildMimeTree = function() {
+  var rootNode;
+
+  if (!this.content.attachments || this.content.attachments.length === 0) {
```

Comment:
I've changed the definition of the function. It can now accept both encrypted and unencrypted content based on whether a MIME message or a PGP/MIME  message is being constructed. 
Does that sound alright?